jina/proto/jina.proto

```diff
@@ -309,6 +310,19 @@ message RequestProto {
         repeated DocumentProto groundtruths = 2; // the groundtruth document you want to evaluate it with
     }
 
+   /**
+     * Represents a dump request
+     */
+    message DumpRequestProto {
+        enum DumpTypes {
+            DEFAULT = 0;
+            NUMPY = 1;
+        }
+        string path = 1; // path for the index
+        int32 shards = 2; // nr of shards in the dump
+        repeated DumpTypes formats = 3; // the formats for which to optimize
```

Comment:
right now, where is this actually used?

Reply:
It should be in the Request, because the user knows what formats the importing Executors supports. The format is about them, not the exporter. The exporting Indexer just passes its data to the DumpPersistor, which will translate it.

Reply:
Refactored it, as it was indeed not yet used.

Reply:
but then better remove it.

Reply:
But what is this `DumpPersistor`? Is it something attached to an `Executor`? Is it an `Executor` itself?

Reply:
> but then better remove it.

Why? It will be used in the future.

Reply:
> But what is this `DumpPersistor`? Is it something attached to an `Executor`? Is it an `Executor` itself?

It is a helper class called by an Indexer to dump its data

Reply:
We will see if it is used in the future, I am not so sure it should be sent with the data.

Right now there is no such a pattern.

I have the feeling that DumpPersistor should evolve into a class to be attached to an Executor (both for the Indexer and the Query One).

Also, every person that develops something new in the hub, will require to do a change in the core? and in our proto definition?

Reply:
> Also, every person that develops something new in the hub, will require to do a change in the core? and in our proto definition?

This is only for some formats, like NumpyIndexer. 

We can delete it, if that's what we want. But pretty sure we agreed we'd want to add this sort of dump format optimization.